compiler/parser/parser.go

```diff
@@ -231,3 +231,9 @@ func (p *Parser) IsNotDefMethodToken() bool {
 
 	return p.curToken.Type != token.Ident && !(p.peekToken.Type == token.Dot && (p.curToken.Type == token.InstanceVariable || p.curToken.Type == token.Constant || p.curToken.Type == token.Self))
 }
+
+// IsNotParamsToken ensures correct parameters which means it is not InstanceVariable
+func (p *Parser) IsNotParamsToken() bool {
+
+	return p.curToken.Type == token.InstanceVariable
```

Comment:
I don't think this is just for instance variable. This should be a table lookup like

```go
_, ok := invalidParams[p.curToken.Type]
return ok
```

And that table should at least contain instance variable and constant